src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp

```diff
@@ -230,36 +237,67 @@ void CARMACloudPlugin::CARMAResponseHandler(QHttpEngine::Socket *socket)
 	tcm=updateTags(tcm,"TrafficControlParams","params");
 	tcm=updateTags(tcm,"TrafficControlGeometry","geometry");
 	tcm=updateTags(tcm,"TrafficControlPackage","package");
-	
-	tsm5Message tsm5message;
-	tsm5EncodedMessage tsm5ENC;
-	tmx::message_container_type container;
+	//List of tcm in string format
+	std::list<std::string> tcm_sl = {};
+	try
+    {
+        std::stringstream iss;
+		iss << tcm;  // updated _cloudUpdate tags, using updateTags
+        boost::property_tree::ptree parent_node;
+        boost::property_tree::read_xml(iss, parent_node);
+        auto child_nodes = parent_node.get_child_optional("TrafficControlMessageList");
+		//The tcm response is a list of TCM
+        if (child_nodes)
+        {
+            for (const auto &p : child_nodes.get())
+            {
+                boost::property_tree::ptree tcm_node;
+                tcm_node.put_child(p.first, p.second);
+                std::ostringstream oss;
+                boost::property_tree::write_xml(oss, tcm_node);
+				tcm_sl.push_back(oss.str());
+            }
+        }else{
+			tcm_sl.push_back(iss.str());

```

Comment:
If I understand this correctly you either receive and XML that contains the element `TrafficControlMessageList` which is a list of XML TCMs and you add each to the tcm_sl. If `TrafficControlMessageList` is not present, the xml payload is just a single TCM. Then you just add it to the tcm_sl. If this is correct please leave some inline comments or move this into helper function so you can document what this logic is for.

Reply:
addressed https://github.com/usdot-fhwa-OPS/V2X-Hub/pull/430/commits/de663dc61f9c951c9b03d4a4a82e7111a7cad0e5